sql/scan.go

```diff
@@ -112,31 +106,49 @@ func (n *scanNode) Next() bool {
 			return false
 		}
 
+		var vals valMap
 		if n.primaryKey == nil {
 			// This is the first key for the row, reset our vals map.
-			n.vals = valMap{}
+			vals = valMap{}
```

Comment:
L109 leaves `vals == nil`, this creates the map. This could also be `vals = make(valsMap)`.